File: java/state/DeliveredOrderState.java
package state;

public class DeliveredOrderState implements OrderState {
    private final OrderManagement orderManagement;

    public DeliveredOrderState(OrderManagement orderManagement) {
        this.orderManagement = orderManagement;
    }

    @Override
    public void processOrder() {
        System.out.println("Order cannot be processed because it's already delivered");
    }

    @Override
    public void shipOrder() {
        System.out.println("Order cannot be shipped because it's already delivered");
    }

    @Override
    public void deliverOrder() {
        System.out.println("Order is already delivered!");
    }

    @Override
    public void cancelOrder() {
        System.out.println("Order cannot be cancelled because it's already delivered");
    }
}

File: PHP/Patterns/src/Behavioral/State/DeliveredOrderState.php
<?php

namespace src\Behavioral\State;

class DeliveredOrderState implements OrderState
{
    private OrderManagement $orderManagement;

    /**
     * @param OrderManagement $orderManagement
     */
    public function __construct(OrderManagement $orderManagement)
    {
        $this->orderManagement = $orderManagement;
    }

    function processOrder()
    {
        print_r("This order is already processed... \n");
    }

    function shipOrder()
    {
        print_r("This order is already shipped... \n");
    }

    function deliverOrder()
    {
        print_r("This order is already delivered... \n");
    }

    function cancelOrder()
    {
        print_r("Can't cancel the order at the current state \n");
    }
}